tools/chapel-py/src/python-types.h

```diff
@@ -135,7 +135,9 @@ PyObject* wrapOptional(ContextObject* context, const std::optional<T>& opt) {
   if (opt) {
     return PythonReturnTypeInfo<T>::wrap(context, *opt);
   } else {
-    Py_RETURN_NONE;
+    // In python3.12, Py_None is immortal and this is not needed
+    Py_INCREF(Py_None);
+    return Py_None;
```

Comment:
Duplicated code and comment. Modulo my previous remark ("why not Py_RETURN_NONE"), I suggest moving this into `python-type-helper.h` (or whatever helper header you like).